### PR TITLE
DCOS-51693 Upgrade to Jest 23

### PR DESCRIPTION
**Pull request by TattdCodeMonkey**
## Testing

```bash
$ npm i
$ npm run test
```

## Trade-offs

Jest 24 requires babel v7, this upgrades to Jest 23 which supports Babel v6 & v7. Once this is merged we can see if it helps with the babel v7 upgrade then prioritize upgrading to Jest v24 which is the latest.

## Dependencies

This PR is based on #3812

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -148,7 +148,8 @@
     "http-server": "github:dcos-labs/http-server#e268ac3fc1f01acdd2d9ec037f7dcb5ec76f7c61",
     "image-webpack-loader": "4.6.0",
     "jasmine-reporters": "2.3.2",
-    "jest-cli": "22.4.0",
+    "jest": "23.6.0",
```
Comment (TattdCodeMonkey) [inline package.json]: `ts-jest` gave warnings with just `jest-cli` 🤷‍♂️

**Comment by GeorgiSTodorov** [comment]
Some tests are failing.
Screenshot from 2019-04-12 10-46-15

**Comment by GeorgiSTodorov** [comment]
Just noticed the PR fixing tests, so I deleted my previous comment.

File: plugins/services/src/js/components/ServiceStatusIcon.tsx [hunk 2]
```diff
@@ -19,7 +19,7 @@ import ServiceTree from "../structs/ServiceTree";
 
 const UNABLE_TO_LAUNCH_TIMEOUT = 1000 * 60 * 30; // 30 minutes
 
-const getTooltipContent = (service: Service, content: JSX.Element) => {
+const getTooltipContent = (service: Service | Pod, content: JSX.Element) => {
   const servicePath = encodeURIComponent(service.getId());
```
Comment (GeorgiSTodorov) [inline plugins/services/src/js/components/ServiceStatusIcon.tsx]: Is this change for this PR?

File: plugins/services/src/js/structs/Service.d.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import Item from "#SRC/js/structs/Item";
-import { Status } from "#PLUGINS/services/src/js/structs/ServiceStatus";
+import { Status } from "#PLUGINS/services/src/js/constants/ServiceStatus";
```
Comment (GeorgiSTodorov) [inline plugins/services/src/js/structs/Service.d.ts]: This change, too?

File: plugins/services/src/js/components/ServiceStatusIcon.tsx (hunk 2 above)
Comment (TattdCodeMonkey) [reply plugins/services/src/js/components/ServiceStatusIcon.tsx]: The first commit is from #3812, now that it’s merged I’ll rebase this.

File: src/js/events/__tests__/MetronomeClient-test.ts [hunk 4]
```diff
@@ -338,6 +337,7 @@ describe("MetronomeClient", () => {
     it(
       "sends 2 requests if schedule is present",
       observe(() => {
+        mockRequest.mockReturnValue(of({}));
```
Comment (nLight) [inline src/js/events/__tests__/MetronomeClient-test.ts]: why do we need more of those?

Comment (TattdCodeMonkey) [reply src/js/events/__tests__/MetronomeClient-test.ts]: I initially had add these to each test, then added it to the `beforeEach` and missed removing them from the tests.

These extra ones can be removed.

**Comment by GeorgiSTodorov** [comment]
Retriggering tests.

**Pull request merged**